mozilla-release/toolkit/mozapps/extensions/internal/XPIProvider.jsm

```diff
@@ -7953,6 +7957,7 @@ Object.assign(SystemAddonInstallLocation.prototype, {
       logger.warn(`System add-on ${aAddon.id} isn't restartless.`);
       return false;
     }
+#endif
```

Comment:
I just have a suspicion, that an update will fail.
Maybe that's why Mozilla put these restrictions here.